Sources/Core/ObjectiveCDictionaryExtension.swift

```diff
@@ -0,0 +1,133 @@
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                ]
+                })
+                ]}
+        case .string(format: .none),
+             .string(format: .some(.email)),
+             .string(format: .some(.hostname)),
+             .string(format: .some(.ipv4)),
+             .string(format: .some(.ipv6)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: "+propIVarName + " forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .string(format: .some(.uri)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: ["+propIVarName + " absoluteString] forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .string(format: .some(.dateTime)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: [[NSValueTransformer valueTransformerForName:\(dateValueTransformerKey)] transformedValue:\(propIVarName)]  forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .enumT(.integer(_)):
+            return "[dict setObject: [NSNumber numberWithInteger:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+        case .enumT(.string(_)):
+            return "[dict setObject: "+enumToStringMethodName(propertyName: param, className: self.className) + "(\(propIVarName))" + " forKey: @\"" + param + "\" ];"
+        case .array(itemType: _):            
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: "+propIVarName + " forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+
+        case .map(valueType: _):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: "+propIVarName + " forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .oneOf(types: _):
```

Comment:
`oneOf` is an algebraic data type (ADT) reference so we need to actually add  `dictionaryRepresentation` to those generated classes. Once we have that method available on ADT classes we can utilize it here